### PR TITLE
Summon Magic now functions like summon guns

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Costs points to use, causes 25% of victims to become antags, and those antags are tasked with stealing the shit that just spawned.

Removed some unused code, and added the one use cap to summon guns as well.

Reasoning behind this:

-Getting a ton of magical loot and then being unallowed to use it was pretty boring. What good is getting a magic weapon if I get banned for using?

-Getting a ton of magical loot but having no antags causes enough cause to make the round unplayable as normal (everything is broke into or horseheaded or turned into carp), but not enough cause to actually cause the round to end (especially with healing wands fixing everyone). This leads to the round being trapped in an awful purgatory where there is no reason to end it, but too messy to do jobs etc.

-Getting bonus points for the book encouraged the wizard to buy this+events (or summon guns) for free, increasing the number of wizards who would spam it.

### CODE REVIEW

**Comment by KorPhaeron** [comment]
@Incoming5643

**Comment by AnonymousNow** [comment]
What was wrong with the previous usage, where it gave you a charge but made no antagonists?

**Comment by lzimann** [comment]
it was boring

**Comment by Incoming5643** [comment]
Fine by me, the challenges menu was only ever added as a compromise to keep summon spells alive.

I think the people who hated said spells aren't still around. (Feel free to prove me wrong!)

**Comment by Niknakflak** [comment]
I still hate them :)

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -704,18 +704,27 @@ var/global/list/possible_items_special = list()
...
 		return 0
-	var/guncount = 0
+	var/stolen_count = 0
 	var/list/all_items = owner.current.GetAllContents()	//this should get things in cheesewheels, books, etc.
-	for(var/obj/I in all_items) //Check for guns
-		if(istype(I, /obj/item/weapon/gun))
-			guncount++
-	if(guncount >= 5)
+	for(var/obj/I in all_items) //Check for wanted items
+		if(is_type_in_list(I, wanted_items))
```
Comment (RemieRichards) [inline code/game/gamemodes/objective.dm]: *loud cough*

Comment (KorPhaeron) [inline code/game/gamemodes/objective.dm]: is this where I get to learn about typecaches or whatever

Comment (RemieRichards) [inline code/game/gamemodes/objective.dm]: yes. for you though it's as easy as:
```
/the/objective/New()
    ...()
    wanted_items = typecacheof(wanted_items)
```
and then using ```is_type_in_typecache(I, wanted_items)```

**Comment by Jordie0608** [comment]
Tbh I thought this was already the case.

**Comment by lzimann** [comment]
It used to be like this when this spell was first made, then someone changed for some stupid reason

**Comment by Nanotransistor** [comment]
Good shit.

**Comment by KorPhaeron** [comment]
@RemieRichards

**Pull request merged**